### ISSUE TITLE
Ability to toggle dark mode off without fiddling with code

### ISSUE THREAD

**Issue by YosypC** [gaphor/gaphor#2115]
Disclaimer: total noob here.

Without the " background-color: rgb(255, 255, 255); " line in the Style Sheet, I see no way of reverting back to light mode. I think that some users would like light mode by default, so a convenient command to switch would be nice to have.

**Comment by amolenaar** [comment]
Dark and light mode depend on your OS settings. On linux (GNOME) they can be toggled, on Windows and macOS the mode is set at application startup. This works fine with me if I clean the style sheet.

You want to switch to dark/light mode independent on the OS settings?

**Comment by YosypC** [comment]
> You want to switch to dark/light mode independent on the OS settings?

Correct

**Comment by amolenaar** [comment]
@YosypC Which OS are you on?

It's doable, there are some other apps allowing you to set the color scheme independent of the system scheme.

Would you expect the whole UI to change, or just the diagrams?

**Comment by YosypC** [comment]
> @YosypC Which OS are you on?
> 
Windows 10
> It's doable, there are some other apps allowing you to set the color scheme independent of the system scheme.
> 
Thanks for the workaround, but an external app shouldn't be required to change a setting of a program, any software has both auto and selectable option.
> Would you expect the whole UI to change, or just the diagrams?
>
The diagram or both (more choices are better), but wasn't the UI already black before?

**Comment by amolenaar** [comment]
> Thanks for the workaround, but an external app shouldn't be required to change a setting of a program, any software has both auto and selectable option.

No, I mean there are code examples available on how to make it work :wink:.

> The diagram or both (more choices are better), but wasn't the UI already black before?

The UI is only black if you use dark-mode on your OS.

**Comment by YosypC** [comment]
> No, I mean there are code examples available on how to make it work 😉.
Aaaahh that's clearer, now. I wish I had the skills and time to implement them :(

> The UI is only black if you use dark-mode on your OS.
I understand. Thank you.

**Comment by danyeaw** [comment]
I think apps should respect system-wide settings by default, which is what we are currently doing. If on Windows you go to Settings > Personalization > Colors and select Dark, you get Dark UI and diagrams. If you set it to Light, you get light UI and diagrams.

While adding an option to override the system-wide settings in possible, I think that it is a lower priority feature. If someone really wants it and wants to submit a PR, I'm all for it though. :smile: 

What @amolenaar is talking about is changing the CSS so that your diagrams are always black (if that's what you want):

```
diagram {
  background-color: #343131;
}

* {
  color: white;
  text-color: white;
}
```
More instructions for this are here: https://docs.gaphor.org/en/latest/style_sheets.html

**Comment by YosypC** [comment]
> I think apps should respect system-wide settings by default, which is what we are currently doing.
> 
Which is a great idea. Automatism that eliminates duplication (in this case, actions) is a great feature. However, sometimes it fails to account for user's micro customization. It can also hide from them possible features and choices, especially if they are behind code (one could not realize there would be a light theme if they didn't read that part of the documentation).

> While adding an option to override the system-wide settings in possible, I think that it is a lower priority feature. 😄
> 
That's absolutely fair. The workaround is easy to implement.

**Comment by amolenaar** [comment]
> However, sometimes it fails to account for user's micro customization.

Since we're running a small project, we like to limit the room for such optimizations, since they can quite easily turn into a source of errors :wink:.

It may be worth having a toggle that disables dark/light mode support completely. Then the diagrams look the same as if they are rendered to an image: diagram exports do not have a dark/light mode.

**Comment by danyeaw** [comment]
Combining what @vanillajonathan is requesting:

I would like a color picker in the property editor to easily be able to assign a background color to a node such as a class or a component.

**Comment by amolenaar** [comment]
Ehhm.. My interpretation is that @vanillajonathan's request is about changing the color of elements in the diagram. This request is about application dark/light mode. I think they should be treated separately.

**Comment by amolenaar** [comment]
Can this issue be closed, or is there any action we have to take?

**Comment by danyeaw** [comment]
@YosypC thanks again for submitting this idea. We won't be implementing this at this time for the reasons previously stated.

**Comment by danyeaw** [comment]
This is now fixed by #2756 and should be out soon in the next release.
